app/controllers/users_controller.rb

```diff
@@ -10,72 +12,50 @@ def create
       session[:user_id] = @user.id
       flash[:success] = "Welcome to Redrum Nursery," \
         " #{@user.first_name} #{@user.last_name}!"
-
       redirect_to dashboard_path
     else
       flash.now[:warning] = @user.errors.full_messages.join(". ")
-
       render :new
     end
   end
 
   def show
-    unless current_user
-      render file: "public/404.html", status: :not_found, layout: true
-    end
   end
 
   def edit
-    if current_user
-      @user = current_user
-    else
-      render file: "public/404.html", status: :not_found, layout: true
-    end
+    @user = current_user
+    @billing = @user.addresses.billing.last
+    @shipping = @user.addresses.shipping.last
   end
 
   def update
-    @user = User.find(params[:id])
-    billing = @user.addresses.billing.last ||
-              Address.new(user_id: @user.id, type_of: 0)
-    shipping = @user.addresses.shipping.last ||
-               Address.new(user_id: @user.id, type_of: 1)
-
-    if (@user.update_attributes(user_params) &&
-        billing.update_attributes(billing_params) &&
-        shipping.update_attributes(shipping_params)
-       )
-      flash[:success] = "Your account has been updated."
-
-      redirect_to dashboard_path
+    @user = current_user
+    @billing = @user.addresses.billing.last
+    @shipping = @user.addresses.shipping.last
+    
+    if !@user.authenticate(params[:user][:password])
+      flash.now[:warning] =
+        "Invalid password. Please re-enter to update your login info."
+    elsif @user.update(user_params)
+      flash.now[:success] = "Your account has been updated."
     else
       flash.now[:warning] = @user.errors.full_messages.join(". ")
-
-      render :edit
     end
+
+    render :edit
   end
 
   private
 
   def user_params
-    params.require(:user).permit(:first_name,
-                                 :last_name,
-                                 :email,
-                                 :password)
+    params.require(:user)
+          .permit(:first_name, :last_name, :email, :password)
```

Comment:
Place the . on the previous line, together with the method call receiver.<br>Use 2 (not 6) spaces for indenting an expression spanning multiple lines.